packages/runtime/src/widgets/Progress.tsx

```diff
@@ -0,0 +1,138 @@
+import React, { useState, useEffect } from "react";
+import { Progress as AntProgress } from "antd";
+import { WidgetRegistry } from "../registry";
+import type { EnsembleWidgetProps } from "../util/types";
+
+export type ProgressProps = {
+  display?: "linear" | "circular";
+  countdown?: number;
+  styles?: {
+    size?: number;
+    thickness?: number;
+    color?: string;
+  };
+} & EnsembleWidgetProps;
+
+const Progress: React.FC<ProgressProps> = (props) => {
+  const { display, countdown, styles } = props;
+
+  // Determine whether to use linear or circular progress based on the "display" prop
+  const isCircular = display === "circular";
+  const strokeWidth = isCircular && styles?.thickness ? styles.thickness : 6; // Use styles.thickness if available for circular display, otherwise default to 6
+
+  // Calculate the percentage based on the countdown value
+  const [percent, setPercent] = useState(countdown ? 0 : -1);
+
+  useEffect(
+    function settingUpInitialAnimation() {
```

Comment:
its just a way to make sure of if ever get some error then tracking it can be easier because its a named function. Is there any issue with it ?

Reply:
Yes, it's unused code. If you need more code to debug what you're writing, then we should revisit the original implementation to make it more clear and easier to debug instead of adding more code. What are you trying to debug here?

Reply:
actually its being fired when `countdown`  property has some value greater than zero, If i set `countdown` to may 2 , this updates the `percent `state for the progress component so that it gets completed in 2 seconds as we have in ensemble progress widget.

Reply:
How is this being fired? You're only defining a function here and not invoking it.

Reply:
Ah sorry my confusion, I'm so used to seeing anonymous functions being passed into useEffect. Nevermind, you're good.